### ISSUE TITLE
Save project with visualparent

### ISSUE THREAD

**Issue by tokejepsen** [getavalon/core#325]
**Problem**

When having visualparent on assets the saving process errors.

State of inventory when loading project:
```toml
[[shots]]
label = "ep0010"
group = "Episode"
name = "ep0010"

[[shots]]
label = "sq0010"
group = "Sequence"
visualParent = 5b3dd2c0cd5f4c097c405697
name = "ep0010_sq0010"

[[shots]]
label = "sh0010"
group = "Shot"
visualParent = 5b3dd2c0cd5f4c097c405699
name = "ep0010_sq0010_sh0010"
```

Error message when ```python -m avalon.inventory --save```:
```
invalid literal for int() with base 10: '5b3dd2c0cd5f4c097c405697'
```

**Comment by davidlatwe** [comment]
Were there any reason that `visualParent` use `ObjectID`, but not using asset name ?

**Comment by mottosso** [comment]
Mm, asset names aren't guaranteed to be unique.

**Comment by davidlatwe** [comment]
Ah, okay, then I should close the PR #348

**Comment by BigRoy** [comment]
Regarding point 1)

- Currently the project manager actually enforces that asset names are unique. It's just that the asset name for episode 1 would be named: `ep01_sh01` and that for episode two would be `ep02_sh01`. You could still _label_ them nicely by setting the `label` in the data. I do think, as such - a unique name is enforced (albeit not by the database)

Nevertheless I believe the GUI approach to showing the name and linking by the id would be a good solution. However, that does mean it's less user-friendly if you will want to be editing these inventory files manually.

> Long-term however, IDs are more robust and what we should do instead is parse that ID from a GUI, where the name is displayed in place of it. That way, we get the best of both worlds. The Project Manager is a good place to do that, which is exactly what it does.

So basically, I suppose I agree with that.

**Comment by davidlatwe** [comment]
> Long-term however, IDs are more robust ..... The Project Manager is a good place to do that, which is exactly what it does.

Agree with that, too.

Then I think, maybe we could hide `visualParent` for the sake of `.toml` editing ?
Or should we stay this way and expose it for manual ID changing ?

I'd prefer hide it, since it's kind of Project Manager's internal use data.

**Comment by davidlatwe** [comment]
PR #349 can prevent it gets upload :)
